operator/v1alpha1/types_kueue.go

```diff
@@ -0,0 +1,305 @@
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Kueue is the CRD to represent the Kueue operator
+// This CRD defines the configuration that the Kueue
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path=kueue,scope=Cluster
+// +k8s:openapi-gen=true
+// +genclient
+// +genclient:nonNamespaced
+// +kubebuilder:storageversion
+// +kubebuilder:subresource:status
+// +kubebuilder:validation:XValidation:rule="self.metadata.name == 'cluster'",message="Kueue is a singleton, .metadata.name must be 'cluster'"
+type Kueue struct {
+	metav1.TypeMeta `json:",inline"`
+	// metadata for Kueue.
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec holds user settable values for configuration
+	// +required
+	Spec KueueOperandSpec `json:"spec"`
+	// status holds observed values from the cluster.
+	// They may not be overridden.
+	// +optional
+	Status KueueStatus `json:"status,omitempty"`
+}
+
+type KueueOperandSpec struct {
+	operatorv1.OperatorSpec `json:",inline"`
+	// config is the desired configuration
+	// for the Kueue operator.
+	// +required
+	Config KueueConfiguration `json:"config"`
+}
+
+type KueueConfiguration struct {
+	// integrations is a required field that configures the Kueue's workload integrations.
+	// Kueue has both standard integrations, known as job frameworks, and external integrations
+	// known as external frameworks.
+	// Kueue will only manage workloads that correspond to the specified integrations.
+	// +required
+	Integrations Integrations `json:"integrations"`
+	// workloadManagement controls how Kueue manages workloads.
+	// By default Kueue will manage workloads that have a queue-name label.
+	// Workloads that are missing the queue-name will be ignored by Kueue.
+	// This field is optional.
+	// If this field is not specified, Kueue will only manage workloads
+	// that have the queue-name label.
+	// +optional
+	WorkloadManagement *WorkloadManagement `json:"workloadManagement,omitempty"`
+	// gangScheduling controls how Kueue admits workloads.
+	// Gang Scheduling is the act of all or nothing scheduling,
+	// where workloads do not become ready within a certain period, they may be evicted and later retried.
+	// This field is optional.
+	// If this field is not specified, gang scheduling will be disabled.
+	// +optional
+	GangScheduling *GangScheduling `json:"gangScheduling,omitempty"`
+	// preemption is the process of evicting one or more admitted Workloads to accommodate another Workload.
+	// Kueue has classical premption and preemption via fair sharing.
+	// This field is optional.
+	// If this field is not specified, preemption will be set to Classical.
+	// +optional
+	Preemption *Preemption `json:"preemption,omitempty"`
```

Comment:
Should we use terminology like "If this field is not specified this means no opinion and the platform is left to choose a reasonable default, which may change over time" for the optional fields?

This would give us some wiggle room for changing the operator's default behaviors without impacting API guarantees we've made, like default values.

Reply:
The language is deliberately vague, to effectively reserve the right to change this, and show to a user, hey, we always said this could change as our opinions on best practices evolve. The configuration APIs we have are flooded with this phrase, so we probably want to continue that here for consistency

Reply:
So for the optional top level, I should put:

If this field is not specified this means no opinion and the platform is left to choose a reasonable default, which may change over time

For preemption I think I'm okay with this but for the other ones I feel pretty good that we won't change the defaults. 
(GangScheduling will be disabled and I don't see us changing that in the current API).
WorkloadManagement will require a queueName.

Reply:
I'll change it for all our optional fields.

Reply:
Addressed this.